### ISSUE TITLE
Incorrect error code being returned when file sizes are changing

### ISSUE THREAD

**Issue by JRDavisUF** [rclone/rclone#5785]
The error code returned by rclone is 1 when a file is unable to be sync'ed because the file size is changing.  For example:

> dd if=/dev/zero of=file bs=1 count=10240000 &
> rclone sync file file-copy --use-json-log -vv --log-file /tmp/blah.log
> echo $?

It seems like this is the incorrect error code number, perhaps 6 would make more sense.

rclone v1.55.1-DEV
- os/type: linux
- os/arch: amd64
- go/version: go1.15.5
- go/linking: dynamic
- go/tags: none

For more details on system, versions, etc., see: https://forum.rclone.org/t/non-jsoned-output-when-using-use-json-log/27299/9

**Comment by ivandeex** [comment]
Return code 1 is widely used as a generic error indicator on Unix and Windows.
So what's wrong with it?

**Comment by ivandeex** [comment]
Ah you probably mean this piece from docs:
https://rclone.org/docs/#list-of-exit-codes

```
0 - success
1 - Syntax or usage error
2 - Error not otherwise categorised
3 - Directory not found
4 - File not found
5 - Temporary error (one that more retries might fix) (Retry errors)
6 - Less serious errors (like 461 errors from dropbox) (NoRetry errors)
7 - Fatal error (one that more retries won't fix, like account suspended) (Fatal errors)
8 - Transfer exceeded - limit set by --max-transfer reached
9 - Operation successful, but no files transferred
```

@ncw 
Feels like this return code mapper was created long ago and not strictly followed.
Do we have unit tests enforcing it?

**Comment by albertony** [comment]
The exit codes are produced here:

https://github.com/rclone/rclone/blob/25366268fe921608ee68c6dc33b744b80d860abe/cmd/cmd.go#L469-L501

And I guess the issue (if any) is that the message produced here is covered:

https://github.com/rclone/rclone/blob/25366268fe921608ee68c6dc33b744b80d860abe/backend/local/local.go#L999-L1004

**Comment by ncw** [comment]
> And I guess the issue (if any) is that the message produced here is covered:

Maybe that case statement needs an `fserrors.IsNoLowLevelRetry` test too?

**Comment by albertony** [comment]
> Maybe that case statement needs an `fserrors.IsNoLowLevelRetry` test too?

Makes sense. Could only see it used from local backend, when size or timestamp are changing during sync.

Added PR where I suggest setting the following exit code for NoLowLevelRetryError, same as for NoRetryError:

> 6 - Less serious errors (like 461 errors from dropbox) (NoRetry errors)

**Comment by animosity22** [comment]
Is there a PR for this item or more to do? Was searching for a PR and I couldn't find anything to link.

**Comment by albertony** [comment]
Is it not #5791?

**Comment by animosity22** [comment]
Yep! Thanks as I was searching and apparently not awake enough to put in the right search. I'd say safe to close this out and once that PR gets committed, it solves this item since it's linked now.

**Comment by albertony** [comment]
I thought it was referenced already, above:

> albertony linked a pull request that will close this issue on 6 Nov

> I'd say safe to close this out and once that PR gets committed, it solves this item since it's linked now.

Yes, that PR will solve it - either as is or it will be updated as necessary to do so. If you want to close this issue before rather than after the PR is merged, is indifferent to me. 👍

**Comment by animosity22** [comment]
Ha, you are double correct. I swear I didn't see it before, but I can see it two places now :) One day when you get older, your eyes will trick you too.

Just trying to clean some things up before the new year so I'll just close it out as @JRDavisUF can follow the PR for tracking purposes as it'll be solved once that is merged.
